### PR TITLE
Fixes #755. Tabular Feature Attribution Tutorial.

### LINKED ISSUE #755: GSoC - Tutorial Project: Feature Attribution
Goal: Create a tutorial for using the Feature Attribution module to determine statistically important features for predictions.

- [x] Draft tutorial in Google Docs
    - Outline major content ideas and flow
    - Collect draft/placeholder images and write up descriptions for them
    - Expand outline to include full-length content
    - Mark any callout sections (see blue insets in [TCAV tutorial](https://pair-code.github.io/lit/tutorials/tcav/))
    - Add any references (i.e., [footnotes](https://www.markdownguide.org/extended-syntax/#footnotes)) and links to the Doc
- [x] Generate final images for tutorial
- [x] Convert Doc and images to a PR
    - Convert tutorial Doc to Markdown
    - Add image files and embed in markdown
    - Build and test website locally 
    - Submit for code review

### PR DESCRIPTION
**Pull request by iamshnoo**
- Changed ```new.css``` to handle info box text ul
- Tutorial contents

### CODE REVIEW

File: website/src/tutorials/tab-feat-attr.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+{% include partials/info-box title: 'Filtering out incomplete data points',
+  text: "Kernel SHAP computes feature importance relative to a pseudo-random
+  sample of the dataset. The sample size is set with the slider, and the samples
+  are drawn from either the current selection (i.e., a subset of the data that
+  were manually selected or are included as part of a slice) or the entire
+  dataset. When sampling from the current selection, the sample size can have
+  interesting edge cases:
+
+- If the sample size is 0, LIT samples the “sample size” number of data points from the entire dataset.
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:71]:
```suggestion
- If the selection is empty, LIT samples the “sample size” number of data points from the entire dataset.
```

File: website/src/tutorials/tab-feat-attr.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+- If the sample size is larger than the selection, then LIT computes SHAP for the entire selection and only that selection.
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:72]:
```suggestion
- If the sample size is zero or larger than the selection, then LIT computes SHAP for the entire selection and only that selection.
```

File: website/src/tutorials/tab-feat-attr.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+a feature’s influence. A reddish hue indicates negative attribution for that
+particular feature and a bluish hue indicates positive attribution. The deeper
+the color the stronger its influence on the predictions.
+
+{% include partials/info-box title: 'Interpreting salience polarity',
+  text: "Intuitively, a positive influence score (attribution) for a training
+example means that if this example was removed from the training dataset, we
+expect a drop in model confidence in predictions on the test set. Similarly,
+a negative score would correspond to an increase in the model's confidence
+in the prediction."%}
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:85]:
```suggestion
{% include partials/info-box title: 'Interpreting salience polarity',
  text: "Salience is always relative to the model’s prediction of one class.
Intuitively, a positive attribution score for a feature of an example means that
 if this feature was removed we
expect a drop in model confidence in the prediction of this class. Similarly,
removing a feature with a negative score would correspond to an increase in
the model's confidence in the prediction of this class."%}
```

Updated to match the same semantics as the salience maps tutorial.

File: website/src/tutorials/tab-feat-attr.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+SHAP values are computed per feature per example, from which LIT computes the
+mean, min, median, and max values from the analyzed examples. The min and max
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:88]:
```suggestion
mean, min, median, and max feature values across the examples. The min and max
```

File: website/src/tutorials/tab-feat-attr.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+the mean and the median can be used to gain more insights about the
+distribution. All of this enables statistical comparisons and will be enhanced
+in future releases of LIT with newer features like histograms to be included
+just for this purpose.
+
+Each of the columns in the table can be sorted using the up (ascending) or down
+(descending) arrow symbols in the column headers. The table is sorted in
+ascending alphabetical order of input feature names (field) by default. If there
+are many features in a dataset this space will get crowded, so LIT offers a
+search button for each of the columns to look up a particular feature or value
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:99]:
```suggestion
filter button for each of the columns to look up a particular feature or value
```

File: website/src/tutorials/tab-feat-attr.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-3.png',
+  caption: 'TODO'%}
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-4.png',
+  caption: 'TODO'%}
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-5.png',
+  caption: 'TODO'%}
+
+#### **Faceting & Binning of numerical features**
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:111]:
```suggestion
#### **Faceting & Binning of Features**
```

File: website/src/tutorials/tab-feat-attr.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+  caption: 'TODO'%}
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-10.png',
+  caption: 'TODO'%}
+
+### **Side-by-side comparison : Salience Maps Vs Tabular Feature Attribution**
+
+The Feature Attribution module works well in conjunction with other modules. In
+particular, we are going to look at the Salience Maps module which allows us to
+enhance our analysis. Salience Maps work on one data point at a time whereas we
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:165]:
```suggestion
enhance our analysis. Salience Maps work on one data point at a time, whereas we
```

File: website/src/tutorials/tab-feat-attr.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+same when a single data point is selected.
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-11.png',
+  caption: 'TODO'%}
+
+#### **A slice of 5 random data points**
+
+LIT uses a [complex selection model](https://github.com/PAIR-code/lit/blob/main/documentation/ui_guide.md#datapoint-selections)
+and different modules react to it differently. Salience Maps only care about the
+primary selection (the data point highlighted deep in the data table) in a slice
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:192]:
```suggestion
primary selection (the data point highlighted in a deep cyan hue in the data table) in a slice
```

File: website/src/tutorials/tab-feat-attr.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+  in the Feature Attribution module. Another thing to note is that the Salience
+  Maps module supports comparison between a pinned datapoint and the primary
+  selection, so we can do the above comparisons in a pair-wise manner as
+  well."%}
+
+As we can see in this example, where we run both modules on a slice of 5
+elements, the Salience Maps module is only providing its output for the primary
+selection (data point 0), whereas the Tabular Feature Attribution module is
+providing values for the entire selection by enabling the checkbox of “Show
+attributions for selection”. This is interesting because we can use the salience
```
Comment (RyanMullins) [inline website/src/tutorials/tab-feat-attr.md:209]:
```suggestion
attributions for selection”. This allows us to use the salience
```

**Review by RyanMullins — APPROVED** [review]

File: website/src/tutorials/tab-feat-attr.md [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+takeaways: "Learn how to use the Kernel SHAP based Tabular Feature Attribution module in LIT."
+---
+
+## Tabular Feature Attribution
+
+{% include partials/link-out link: "../../demos/penguins.html", text: "Explore this demo yourself." %}
+
+Or, run your own with [`examples/penguin_demo.py`](https://github.com/PAIR-code/lit/blob/main/lit_nlp/examples/penguin_demo.py)
+
+LIT supports many techniques like salience maps and counterfactual generators for text data. But what if you have a tabular dataset? You might want to find out which data features are most relevant in the model’s predictions. LIT provides a Feature Attribution module, which is currently supported for [tabular datasets](https://github.com/PAIR-code/lit/wiki/components.md#tabular-data) only, to enable important feature identification. This tutorial provides a walkthrough for this module within LIT, on the [Palmer Penguins dataset](https://allisonhorst.github.io/palmerpenguins/).
```
Comment (jameswex) [inline website/src/tutorials/tab-feat-attr.md:23]: "identification of important features" reads better than "import feature identification"

File: website/src/tutorials/tab-feat-attr.md [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+2. The **prediction key** selects the model output value for which influence is computed. Since the penguin mode only predicts one feature, species, this is set to species and cannot be changed. If a model can predict multiple values in different fields, for example predicting species and island or species and sex, then you could change which output field to explain before clicking Apply.
+3. The **heatmap toggle** can be enabled to color code the SHAP values.
+4. The **facets button** and **show attributions for selection checkbox** enable
+   conditionally running the Kernel SHAP interpreter over subsets of the data.
+   We will get into the specifics of this with an example later on in this
+   tutorial.
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-1.png',
+  caption: 'TODO'%}
```
Comment (jameswex) [inline website/src/tutorials/tab-feat-attr.md:53]: be sure to add captions

Comment (RyanMullins) [reply website/src/tutorials/tab-feat-attr.md:53]: Leaving captions as TODOs intentionally as we're going to 1) convert some images to GIFs and need to capture the narrative of those GIFs, and 2) annotate the static images using the typical LIT callout styles and will need to write the captions with those annotations in mind.

File: website/src/tutorials/tab-feat-attr.md [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+
+{% include partials/inset-image image: '/assets/images/tab-feat-attr-image-2.png',
+  caption: 'TODO'%}
+
+#### **A Simple Use Case : Feature Attribution for 10 samples**
+
+To get started with the module, we set sample size to a small value, 10, and
+start the SHAP computation with heatmap enabled.
+
+{% include partials/info-box title: 'Filtering out incomplete data points',
```
Comment (jameswex) [inline website/src/tutorials/tab-feat-attr.md:63]: update title

**Review by jameswex — APPROVED** [review]

**Review by RyanMullins — APPROVED** [review]

**Pull request merged**